client/src/LandingPages/SignupPage/SignupPage.jsx

```diff
@@ -3,62 +3,133 @@ import './SignupPage.css';
+    password: '',
+    confirmPassword: '',
   });
 
+  const [errorMsg, setErrorMsg] = useState('')
+  const [successMsg, setSuccessMsg] = useState('')
+
+
+  // Add more form fields as needed here:
+  const formFields = [
+    {
+      name: 'firstName',
+      label: 'First Name',
+      type: 'text',
+      id: "signup-firstName",
+      htmlFor: "signup-firstName",
+      errorMessage: "First name needs to be 2 to 20 characters long and shouldn't contain special characters.",
+      pattern: "^[A-Za-z]{2,20}$",
+      icon: <TbPencilCheck />,
+      required: true,
+    },
+    {
+      name: 'lastName',
+      label: 'Last Name',
+      type: 'text',
+      id: "signup-lastName",
+      htmlFor: "signup-lastName",
+      errorMessage: "Last name needs to be 2 to 20 characters long and shouldn't contain special characters.",
+      pattern: "^[A-Za-z]{2,20}$",
+      icon: <TbPencilCheck />,
+      required: true,
+    },
+    {
+      name: 'username',
+      label: 'Username',
+      type: 'text',
+      id: "signup-username",
+      htmlFor: "signup-username",
+      errorMessage: "Username should be 3-20 characters. Only letters, numbers and underscores are allowed.",
+      pattern: "^[A-Za-z0-9_]{3,20}$",
+      icon: <FaUserPlus />,
+      required: true,
+    },
+    {
+      name: 'email',
+      label: 'Email',
+      type: 'email',
+      id: "signup-email",
+      htmlFor: "signup-email",
+      errorMessage: "Please use a valid email address.",
+      icon: <MdAlternateEmail />,
+      required: true,
+    },
+    {
+      name: 'password',
+      label: 'Password',
+      type: 'password',
+      id: "signup-password",
+      htmlFor: "signup-password",
+      errorMessage: "Password should be 6-20 characters. Include letters, numbers and at least one special character.",
+      pattern: "^(?=.*[A-Za-z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]{6,20}$",
+      icon: <RiLockPasswordFill />,
+      required: true,
+    },
+    {
+      name: 'confirmPassword',
+      label: 'Confirm Password',
+      type: 'password',
+      id: "signup-confirmPassword",
+      htmlFor: "signup-confirmPassword",
+      errorMessage: "Passwords do not match",
+      pattern: inputValue.password,
+      icon: <RiLockPasswordFill />,
+      required: true,
+    },
+  ]
+  
   const handleChange = (e) => {
-    const { name, value } = e.target;
-    setInputValue({
-      ...inputValue,
-      [name]: value
-    });
+    setInputValue({...inputValue, [e.target.name]: e.target.value});
   };
 
-  //CHANGE THIS FUNCTION FOR SIGNUP TASK #2
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    console.log('submit');
-  };
+  async function handleSubmit(evt) {
+    evt.preventDefault();
+    try {
+      const user = await authService.signUp(inputValue);
+      setUser(user);
+      setSuccessMsg('Sign Up Successful', user);
+    } catch (err) {
+      console.log(err);
+      setErrorMsg('Sign Up Failed - Try Again');
+    }
+  }
```

Comment:
If a user is able to successfully create their profile, the page should automatically navigate to the home/landing page (or login page even); it shouldn't stay on the signup page.